edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/Metric.cs

```diff
@@ -33,7 +34,7 @@ public int GetMetricKey()
             // when upgraded to .net standard 2.1: https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=netstandard-2.1
             int hash = 17;
             hash = hash * 31 + this.Name.GetHashCode();
-            hash = hash * 31 + this.Tags.GetHashCode();
+            hash = hash * 31 + JsonConvert.SerializeObject(this.Tags).GetHashCode();
```

Comment:
why do json convert? just concatenate tags instead.

Reply:
Add how can you ensure the order of tags is same all the time when calling GetHashCode

Reply:
I didn't think JsonConvert.SerializeObject added enough overhead to be worth implementing manually. I could be wrong about this though. 

For the tag ordering, the original implementation would yield different hash codes for different tag orders, which I agree can be improved. I should be calling GetHashCode() on the dictionary object anyways though which I will change.